linter/src/main/java/linter/App.java

```diff
@@ -14,47 +14,34 @@ public static void main(String[] args) {
     }
 
     // scanning came from https://docs.oracle.com/javase/tutorial/essential/io/scanning.html
-    public static void getAliceInWonderland() {
+    public static String javascriptLinter() {
         Scanner s = null;
-
-        int timesAliceIsMentioned = 0;
+        StringBuilder sb = new StringBuilder();
+        int lineNumber = 0;
 
         try {
             s = new Scanner(new BufferedReader(new FileReader("./src/main/resources/gates.js")));
-            while (s.hasNext()) {
-                if (s.next().contains("Alice")) {
-                    timesAliceIsMentioned++;
-                }
-            }
-
-            System.out.println(String.format("Alice is mentioned %d times.", timesAliceIsMentioned));
-
-        } catch (FileNotFoundException e) {
-            System.out.println("the file was not found");
-        }
-    }
-
-
-
 
+            while (s.hasNextLine()) {
+                String line = s.nextLine();
 
+                lineNumber++;
 
+                if (!line.isEmpty()) {
+                    char lastChar = line.charAt(line.length() - 1);
 
-    public static void javascriptLinter() {
-        Scanner s = null;
-
-        try {
-            s = new Scanner(new BufferedReader(new FileReader("./src/main/resources/gates.js")));
-
-
-            while (s.hasNextLine()) {
-                System.out.println(s.nextLine());
+                    if (lastChar != ';' && lastChar != '{' && lastChar != '}' && !line.contains("if") && !line.contains("else")) {
```

Comment:
When you have long conditionals like this it's common to put each conditional statement on its own line for readability:
```
if (lastChar != ';' 
     && lastChar != '{' 
     && lastChar != '}' 
     && !line.contains("if") 
     && !line.contains("else")) {
```